### DISCUSSION TITLE
Features from the Browser not working: SCM, Run and Debug, Github

### DISCUSSION THREAD

**Discussion by gdevacc12** [community/community#17375] [Codespaces]
I tried the Codespaces quick start but some features using the web browser did not work.

1. The SCM (git): the project source code management did not work. It was just stuck at the "Initialise Repository" button, which did nothing.
2. Run and Debug: clicking the "Javascript Debug Terminal" button, the first time showed "Activating extensions" in the bottom status bar but then nothing else happened. Clicking again even after changing views also did nothing. Refreshing the browser and clicking "Javascript Debug Terminal" button again showed the "Activating extensions" but the behaviour repeated.
3. Github: stuck on "loading", could not see status or pull, push state.
4. Generally switching between the features above was slow and the first time loading the view was slow. This could be network issue as there were a couple of disconnect/reconnects. I was trying from Germany.

From Visual code running locally on my PC, once Visual Code was authenticated (naturally), then it seemed like all the features worked fine. The VC Run/Debug default "Launch Chrome against localhost" which caught me out initially. Switching to "Run Script:start" everything worked fine.

Windows 11 21H2 Build: 22000.675 x64,  Chrome: Version 101.0.4951.67 (Official Build) (64-bit), Visual Code: 1.67.2,

**Comment by jshorty** [comment]
👋 Hi from Codespaces and thank you for the report. From your description, I suspect what happened is that the codespace ran into performance issues cloning down the repository— there was also an [incident](https://www.githubstatus.com/incidents/zhtplv7zd052) close to the time of this report, including degraded performance for Git Operations. However, if you were able to see and interact with the file tree within the browser editor during that time, that would rule this out. Glad to hear things worked well from VS Code on the desktop.

Please let us know if you consistently see these problems from the web editor! We have codespaces running on infrastructure in West Europe, so unusually high latency would be unexpected.

**Reply by gdevacc12** [reply]
Thanks for the prompt response. I tried again and the same behaviour persists from the browser ( but local Visual Code works fine).
e.g.:
- Source Control just shows the "Initialise Repository" button,
- Run & Debug the "JavaScript Debug Terminal" button appears not to do anything.
- GitHub is stalled on "Loading".

**Reply by jshorty** [reply]
Sorry to hear you're still encountering problems. Do you mind sharing the web editor URL for one of the codespaces for which you're seeing this? It's a safe URL to share, as only you can authenticate and connect to it, but it will allow us to better look for errors in telemetry. I'll file an internal issue with our team to investigate further.

**Comment by gdevacc12** [comment]
Thanks for pursuing the problem.
The URL is:
https://github.com/codespaces/gdevacc12-codespacequickstart-q4v6rgp3x6gv
